foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/platform/database/Oracle8Platform.java

```diff
@@ -14,12 +14,341 @@
+            // for the writing
+            // for the time being, simply use Oracle api.
+            try {
+                final Method putStringMethod = clobClass.getMethod("putString", int.class, String.class);
+                putStringMethod.invoke(clob, 1, (String) value);
+
+                // impose the locallization
+                final Method clobLengthMethod = clobClass.getMethod("length");
+                Long length = Long.valueOf((long) clobLengthMethod.invoke(clob));
+                session.log(SessionLog.FINEST, SessionLog.SQL, "write_CLOB", length, field.getName());
+            } catch (Exception e) {
+                // TODO: Emit Warning
+            }
+        } else {
+            // do nothing for now, open to BFILE or NCLOB types
+        }
+    }
+
+    /**
+     * INTERNAL: Used in writeLOB method only to identify a BLOB
+     */
+    protected boolean isBlob(Class type) {
+        return ClassConstants.BLOB.equals(type);
+    }
+
+    /**
+     * INTERNAL: Used in writeLOB method only to identify a CLOB
+     */
+    protected boolean isClob(Class type) {
+        return ClassConstants.CLOB.equals(type);
+    }
+
+    /**
+     * INTERNAL: Indicates whether app. server should unwrap connection to use
+     * lob locator.
+     */
+    public boolean isNativeConnectionRequiredForLobLocator() {
+        return true;
+    }
+
+    /**
+     * PUBLIC: Set if the locator is required for the LOB write. The default is
+     * true. For Oracle thin driver, the locator is recommended for large size (
+     * >4k for Oracle8, >5.9K for Oracle9) BLOB/CLOB value write.
+     */
+    public void setShouldUseLocatorForLOBWrite(boolean usesLocatorForLOBWrite) {
+        this.usesLocatorForLOBWrite = usesLocatorForLOBWrite;
+    }
+
+    /**
+     * PUBLIC: Return if the locator is required for the LOB write. The default
+     * is true. For Oracle thin driver, the locator is recommended for large
+     * size ( >4k for Oracle8, >5.9K for Oracle9) BLOB/CLOB value write.
+     */
+    public boolean shouldUseLocatorForLOBWrite() {
+        return usesLocatorForLOBWrite;
+    }
+
+    /**
+     * PUBLIC: Return the BLOB/CLOB value limits on thin driver. The default
+     * value is 0. If usesLocatorForLOBWrite is true, locator will be used in
+     * case the lob's size is larger than lobValueLimit.
+     */
+    public int getLobValueLimits() {
+        return lobValueLimits;
+    }
+
+    /**
+     * PUBLIC: Set the BLOB/CLOB value limits on thin driver. The default value
+     * is 0. If usesLocatorForLOBWrite is true, locator will be used in case the
+     * lob's size is larger than lobValueLimit.
+     */
+    public void setLobValueLimits(int lobValueLimits) {
+        this.lobValueLimits = lobValueLimits;
+    }
+
+    /**
+     * INTERNAL: Used by Oracle platforms during reading of ResultSet to free
+     * temporary LOBs.
+     */
+    public void freeTemporaryObject(Object value) throws SQLException {
```

Comment:
Was this method required to be ported as part of the LOB Locator support? It looks like it was added under this issue: https://bugs.eclipse.org/bugs/show_bug.cgi?id=294578. If not, then we don't need to introduce the reflection at this time.

Reply:
I'll remove it.